### ISSUE TITLE
Reduce code complexity below 16 and fix up long if statementws

### ISSUE THREAD

**Issue by isaacmg** [AIStream-Peelout/flow-forecast#301]
It is no secret that the complexity of some functions has gotten out of hand. We need to work to reduce complexity and readibility of long functions. Of particular concern are long if else statements particulary those that are repeated many time throughout the code.

**Comment by theadammurphy** [comment]
Agreed! It was my first time working with FF yesterday and it was hard to follow along. 

My first thought is just to split the massive functions up into smaller ones with more readable names?
